### DISCUSSION TITLE
How to export the theme function to js?

### DISCUSSION THREAD

**Discussion by IgorDePaula** [tailwindlabs/tailwindcss#5172] [Help] [answered]
How to export the theme function to js? The idea is create a design system function hook like material-ui. I tried import into js from many forms:

```
import theme  from 'tailwindcss'
import {theme} from 'tailwindcss'
import theme from 'tailwindcss/defaultTheme'

theme('spacing.12') => is not a function
theme.theme('spacing.12') => is not a function
```

**Comment by dance2die** [comment, marked as answer]
You can reference resolved configuration values (including the theme): 
https://tailwindcss.com/docs/configuration#referencing-in-java-script


From the linked doc, 

```js
import resolveConfig from 'tailwindcss/resolveConfig'
import tailwindConfig from './tailwind.config.js'

const fullConfig = resolveConfig(tailwindConfig)

fullConfig.theme.width[4]
// => '1rem'

fullConfig.theme.screens.md
// => '768px'

fullConfig.theme.boxShadow['2xl']
// => '0 25px 50px -12px rgba(0, 0, 0, 0.25)'
```

**Reply by IgorDePaula** [reply]
OOOOOOhhh, nice, Thank you very much.

**Reply by sara-jegorova** [reply]
But if I'm using create-react-app I'm not able to import `tailwind.config.js`, as it doesn't support imports outside of the `src` folder. I'm not sure what's the solution in this case 🤔

**Reply by sara-jegorova** [reply]
For anyone wondering how I fixed this issue, I added babel-plugin-preval plugin using craco to avoid ejecting CRA and then did:

```
const fullConfig = preval`
  const resolveConfig = require('tailwindcss/resolveConfig');
  const tailwindConfig = require('../../tailwind.config');
  module.exports = resolveConfig(tailwindConfig);
`;
```

**Reply by dealer-solutions-gene** [reply]
`resolveConfig()` was [removed](https://tailwindcss.com/docs/upgrade-guide#theme-values-in-javascript) after upgrading to `4.x.x` for anyone finding this in 2025.

> We've removed this in v4 in hopes that people can use the CSS variables we generate directly instead, which is much simpler and will significantly reduce your bundle size.
